### DISCUSSION TITLE
Building your own bootc container image

### DISCUSSION THREAD

**Discussion by dainsco** [bootc-dev/bootc#755] [Q&A] [answered]
Is there an example or any documentation on how you can create your own bootc base container?  What are the requirements of the distro that can be selected?

**Comment by miabbott** [comment, marked as answer]
In the case of Fedora bootc base images, they are built from the repo here - https://gitlab.com/fedora/bootc/base-images

There is some light documentation in that repo about how they are built (and how you can build them yourself), though it could be improved. 

I think a longer term goal should be to have an easier, more supportable way of building your own base images, but currently it appears that you need to do some tinkering.

**Reply by dainsco** [reply]
Thanks for the pointer, I think information is available just quite hard to find.

I there anything in the base distribution that is required by the bootc build?  Would it be feasible to create a base image for RHEL 8 or is RHEL 9 required?

**Reply by KristoferHansson** [reply]
@dainsco : did you find anything valuable, and did you prepare a custom bootc image based on some non bootc ready distro?

**Comment by KristoferHansson** [comment]
For yocto there is the meta-updater, it't does a few other things on top of just ostree adaption, linked to specific tooling related to Uptane, such as aktualizer(agent) and garage(server), but once you dig in to it you can see the rootfs modifcations it does to satisfy ostree's requirements, for example: https://github.com/advancedtelematic/meta-updater/blob/master/classes/image_types_ostree.bbclass#L25, of course there are more things with the kernel, kernel arguments, initramfs etc.

Would like to see some guidance related to bringing the plain ostree adaption to include adapting to the full bootc image requirements.

**Reply by KristoferHansson** [reply]
Remembered some blog post related to adapting your os to ostree and tried to find it again:
https://community.sap.com/t5/open-source-blogs/garden-linux-fellowship-retrospective/ba-p/13579463

**Comment by KristoferHansson** [comment]
Look at https://github.com/bootcrew for a number of distros(currently 10) adapted to and packaged as bootc images
